aten/src/ATen/native/nested/NestedTensorTransformerFunctions.cpp

```diff
@@ -8,6 +8,32 @@
 namespace at {
 namespace native {
 
+Tensor NestedTensor_linear(
+    const Tensor& input,
+    const Tensor& weight,
+    const c10::optional<Tensor>& bias_opt) {
+  auto* nt_input = get_nested_tensor_impl_or_null(input);
+  TORCH_CHECK(nested_tensor_impl_is_contiguous(nt_input));
+  TORCH_CHECK(input.dim() == 3 && weight.dim() == 2);
```

Comment:
Just to check my understanding, the only case when input.dim() != 3 is when input.dim() == 1 and that is when the nested tensor is created from an empty list?

Reply:
A NestedTensor can have any dimension. The dimension of a NestedTensor is 1 greater than that of its constituents (just like a regular torch Tensor). Or put differently, if you unbind a 3 dimension torch Tensor, you get a list of 2 dimensional torch Tensors.

Reply:
I see, thanks!